packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -368,9 +394,38 @@ contract Exchange is
     uint256 updatedStableBucket = exchangeRateNumerator.mul(updatedGoldBucket).div(
       exchangeRateDenominator
     );
+
+    uint256 maxStableBUcketSize = getStableBucketTokenCap();
+
+    // check if the token is bigger than the cap
+    uint256 cappedUpdatedStableBucket = Math.min(maxStableBUcketSize, updatedStableBucket);
+
+    if (cappedUpdatedStableBucket < updatedStableBucket) {
+      // resize gold bucket
+      uint256 cappedUpdatedGoldBucket = exchangeRateDenominator.mul((maxStableBUcketSize)).div(
+        exchangeRateNumerator
+      );
+      return (cappedUpdatedGoldBucket, cappedUpdatedStableBucket);
+    }
+
     return (updatedGoldBucket, updatedStableBucket);
   }
 
+  /**
+   * @notice returns the max size the stableToken can be set durin a bucket update. if the supply of stableToken
+   * is smaller than minSupplyForStableBucketCap, the cap is calculated based on minSupplyForStableBucketCap.
+   */
+  function getStableBucketTokenCap() public view returns (uint256) {
+    uint256 stableTokenSupply = Math.max(IERC20(stable).totalSupply(), minSupplyForStableBucketCap); // assume a miminum of 1M stable tokens supply to start the cap
```

Comment:
Isn't this a somewhat weird situation with let's say `minSupplyForStableBucketCap == 1M` and `maxStableBucketFraction == 0.1`? I.e. the max bucket size is 1M for tokenSupply <1M, but once supply hits 1M, the max is actually 0.1M?

Reply:
In that case for a supply of zero to 1M, you get a cap of `1M*0.1`, then from there you get a cap of `Supply*0.1`. Or at least that was my intention.

Reply:
You are right, I misread the code. The comment led me to believe that the cap doesnt apply until a supply of 1M, so maybe I would reword the comment